client/components/data/purchases/edit-card-details/index.jsx

```diff
@@ -32,7 +32,7 @@ function getStateFromStores( props ) {
 
 function isDataLoading( state ) {
 	return (
-		! state.card ||
+		state.card.isFetching ||
```

Comment:
This fix only addresses part of the issues mentioned in https://github.com/Automattic/wp-calypso/issues/1264. The problem is more deep and need some thinking but I guess this workaround is fine for the time being.